src/main/java/net/snowflake/ingest/streaming/internal/ExternalVolumneManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.ingest.connection.IngestResponseException;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import net.snowflake.ingest.utils.Utils;
+
+class ExternalVolumeLocation {
+  public final String dbName;
+  public final String schemaName;
+  public final String tableName;
+
+  public ExternalVolumeLocation(String dbName, String schemaName, String tableName) {
+    this.dbName = dbName;
+    this.schemaName = schemaName;
+    this.tableName = tableName;
+  }
+}
+
+/** Class to manage multiple external volumes */
+class ExternalVolumeManager<T> implements IStorageManager<T, ExternalVolumeLocation> {
+  // Reference to the external volume per table
+  private final Map<String, StreamingIngestStorage<T, ExternalVolumeLocation>> externalVolumeMap;
```

Comment:
Does this need to be thread safe?

Reply:
It's using conncurrentHashMap, I think it should be ok